apps/prairielearn/src/lib/group-update.js

```diff
@@ -193,56 +170,52 @@ export async function autoGroups(
         job.verbose('Auto generate group settings for ' + assessmentLabel);
         job.verbose(`----------------------------------------`);
         job.verbose(`Fetching the enrollment lists...`);
-        const enrollments = await queryRows(sql.select_enrollments, { assessment_id }, z.string());
-        _.shuffle(enrollments);
-        const numStudents = enrollments.length;
-        job.verbose(`There are ` + numStudents + ' students enrolled in ' + assessmentLabel);
-        job.verbose(`----------------------------------------`);
-        job.verbose(
-          `Processing creating groups - max of ` + max_group_size + ' and min of ' + min_group_size,
+        const studentsToGroup = await queryRows(
+          sql.select_enrolled_students_without_group,
+          { assessment_id },
+          UserSchema,
         );
-        let numGroup = Math.ceil(numStudents / max_group_size);
-        let updateList = [];
-        // fill in the updateList with groupname and uid
-        for (let i = 0; i < numGroup; i++) {
-          let groupName = 'group' + i;
-          for (let j = 0; j < max_group_size; j++) {
-            if (enrollments.length > 0) {
-              let uid = enrollments.pop();
-              updateList.push([groupName, uid]);
-            }
-          }
-        }
-        const result = await callRow(
-          'assessment_groups_update',
-          [assessment_id, updateList, authn_user_id],
-          AssessmentGroupsUpdateResultSchema,
+        _.shuffle(studentsToGroup);
+        const numStudents = studentsToGroup.length;
+        job.verbose(
+          `There are ${numStudents} students enrolled in ${assessmentLabel} without a group`,
         );
-        let errorCount = 0;
-
-        const notExist = result.not_exist_user;
-        if (notExist) {
-          job.verbose(`----------------------------------------`);
-          job.verbose(`ERROR: The following users do not exist. Please check their uids first.`);
-          notExist.forEach((user) => job.verbose(user));
-          errorCount += notExist.length;
-        }
+        job.verbose(`----------------------------------------`);
+        job.verbose(`Processing creating groups - max of ${max_group_size}`);
 
-        const inGroup = result.already_in_group;
-        if (inGroup) {
-          job.verbose(`----------------------------------------`);
-          job.verbose(`ERROR: The following users are already in a group.`);
-          inGroup.forEach((user) => job.verbose(user));
-          errorCount += inGroup.length;
+        // Find a group name of the format `groupNNN` that is not used
+        const groupNumber = await queryOptionalRow(
+          sql.select_unused_group_name,
```

Comment:
I have renamed it in 2c55f84ea284fd8db7667cef07dcf612d84679ef to `select_unused_group_name_suffix`. This is not exactly a group number, so I wanted to avoid confusion.